src/network/protocol/messages/cmpctblock.zig

```diff
@@ -0,0 +1,220 @@
+const std = @import("std");
+const protocol = @import("../lib.zig");
+const Transaction = @import("../../../types/transaction.zig");
+
+const Sha256 = std.crypto.hash.sha2.Sha256;
+const BlockHeader = @import("../../../types/block_header.zig");
+const CompactSizeUint = @import("bitcoin-primitives").types.CompatSizeUint;
+const genericChecksum = @import("lib.zig").genericChecksum;
+
+pub const CmpctBlockMessage = struct {
+    header: BlockHeader,
+    nonce: u64,
+    short_ids: []u64,
+    prefilled_txns: []PrefilledTransaction,
+
+    const Self = @This();
+
+    pub const PrefilledTransaction = struct {
+        index: usize,
+        tx: Transaction,
+    };
+
+    pub fn name() *const [12]u8 {
+        return protocol.CommandNames.CMPCTBLOCK;
+    }
+
+    pub fn checksum(self: *const Self) [4]u8 {
+        return genericChecksum(self);
+    }
+
+    pub fn deinit(self: *Self, allocator: std.mem.Allocator) void {
+        allocator.free(self.short_ids);
+        for (self.prefilled_txns) |*txn| {
+            txn.tx.deinit();
+        }
+        allocator.free(self.prefilled_txns);
+    }
+
+    pub fn serializeToWriter(self: *const Self, w: anytype) !void {
+        comptime {
+            if (!@hasDecl(BlockHeader, "serializeToWriter")) {
+                @compileError("BlockHeader must be serializable");
+            }
+            if (!@hasDecl(Transaction, "serializeToWriter")) {
+                @compileError("Transaction must be serializable");
+            }
```

Comment:
no, you should check for `writeInt`.

BlockHeader and Transaction are types known at comptime. If you were calling a method that is not declared on them, the compilation would fail.

Because we are using anytype for `w` we have to manually check